### PR TITLE
Show percentage switch

### PR DESCRIPTION
**Pull request by andirsun**
fixes: #167

### CODE REVIEW

**Comment by jeremypw** [comment]
Just a tip: If a PR changes the appearance then it is usual to include a screenshot of the changed appearance.

File: src/MainView.vala [hunk 1]
```diff
@@ -62,6 +63,16 @@ public class Power.MainView : Gtk.Grid {
         main_grid.column_spacing = 12;
         main_grid.row_spacing = 12;
 
+        var show_percent_label = new Gtk.Label (_("Show Percentage"));
+        show_percent_label.halign = Gtk.Align.END;
+        show_percent_label.xalign = 1;
```
Comment (cassidyjames) [inline src/MainView.vala:68]: Modern code style and whitespace between 

```suggestion
        var show_percent_label = new Gtk.Label (_("Show Percentage")) {
            halign = Gtk.Align.END,
            xalign = 1
        };

```

File: src/MainView.vala [hunk 2]
```diff
@@ -62,6 +63,16 @@ public class Power.MainView : Gtk.Grid {
...
+        var show_percent_switch = new Gtk.Switch ();
+        show_percent_switch.halign = Gtk.Align.START;
```
Comment (cassidyjames) [inline src/MainView.vala:70]: Modern code style

```suggestion
        var show_percent_switch = new Gtk.Switch () {
            halign = Gtk.Align.START
        };
```

File: src/MainView.vala [hunk 3]
```diff
@@ -62,6 +63,16 @@ public class Power.MainView : Gtk.Grid {
...
+        wingpanel_power_settings.bind ("show-percentage", show_percent_switch, "active", SettingsBindFlags.DEFAULT);
+
+        main_grid.attach (show_percent_label, 0, 0, 1, 1);
+        main_grid.attach (show_percent_switch, 1, 0, 1, 1);
```
Comment (cassidyjames) [inline src/MainView.vala:74]: You can omit the span parameters as they default to 1

```suggestion
        main_grid.attach (show_percent_label, 0, 0);
        main_grid.attach (show_percent_switch, 1, 0);
```

File: src/MainView.vala [hunk 4]
```diff
@@ -85,10 +96,10 @@ public class Power.MainView : Gtk.Grid {
...
             ((DBusProxy)screen).g_properties_changed.connect (on_screen_properties_changed);
 
-            main_grid.attach (brightness_label, 0, 0, 1, 1);
-            main_grid.attach (scale, 1, 0, 1, 1);
-            main_grid.attach (als_label, 0, 1, 1, 1);
-            main_grid.attach (als_switch, 1, 1, 1, 1);
+            main_grid.attach (brightness_label, 0, 1, 1, 1);
+            main_grid.attach (scale, 1, 1, 1, 1);
+            main_grid.attach (als_label, 0, 2, 1, 1);
+            main_grid.attach (als_switch, 1, 2, 1, 1);
```
Comment (cassidyjames) [inline src/MainView.vala:102]: Same as above; as long as we're updating this code, we can drop the implied `1` values.

```suggestion
            main_grid.attach (brightness_label, 0, 1);
            main_grid.attach (scale, 1, 1);
            main_grid.attach (als_label, 0, 2);
            main_grid.attach (als_switch, 1, 2);
```

**Review by cassidyjames — CHANGES_REQUESTED** [review]
Just some code style nits

File: src/MainView.vala [hunk 5]
```diff
@@ -62,6 +63,19 @@ public class Power.MainView : Gtk.Grid {
         main_grid.column_spacing = 12;
         main_grid.row_spacing = 12;
 
+        var show_percent_label = new Gtk.Label (_("Show Percentage")) {
```
Comment (ryonakano) [inline src/MainView.vala:66]:
```suggestion
        var show_percent_label = new Gtk.Label (_("Show percentage:")) {
```

For consistence with other labels in this view

**Comment by ryonakano** [comment]
Now it looks like:

Screenshot from 2021-07-28 08-24-58

**Review by ryonakano — DISMISSED** [review]
Works for me, I felt we may need a redesign using header labels though. But it should be done in another PR.

**Comment by ryonakano** [comment]
I tested in a desktop machine that does not have any batteries and felt we should hide this switch if there are no batteries. So I added codes to check for battery existence.

@andirsun @jeremypw Would you check if this branch still works for you?

**Comment by jeremypw** [comment]
Yeah, the "Single Exit Point" school of coding seems to have gone out of fashion ...

**Comment by ryonakano** [comment]
The variable `present` is not used in the method for other purposes and I think the method is clean enough to notice two returns, so I removed it. Should I keep it as it is? 🤔

**Comment by andirsun** [comment]
Yes its the same behavior.

**Review by ryonakano — APPROVED** [review]
Works as expected―the switch is hidden on the desktop and it appears and works on the laptop. Nice job!

**Pull request merged**